packages/app/features/auth/account-recovery/eoa/RecoverWithEOA.tsx

```diff
@@ -0,0 +1,63 @@
+import { useAccount, useSignMessage } from 'wagmi'
+import type { SignMessageData, SignMessageVariables } from 'wagmi/query'
+import { Button, ButtonText } from '@my/ui'
+import { Provider } from 'app/provider'
+import { OpenConnectModalWrapper } from 'app/utils/OpenConnectModalWrapper'
+import {
+  type ChallengeResponse,
+  type VerifyChallengeRequest,
+  RecoveryOptions,
+} from '@my/api/src/routers/account-recovery/types'
+import type { SignState } from 'app/features/auth/account-recovery/account-recovery'
+
+interface Props {
+  challengeData: ChallengeResponse
+  signState: SignState
+
+  onSignSuccess: (args: VerifyChallengeRequest) => Promise<void>
+  onSignError: () => void
+}
+
+export default function RecoverWithEOA(props: Props) {
+  const { signMessage } = useSignMessage()
+  const { address } = useAccount()
+
+  // https://wagmi.sh/react/api/hooks/useSignMessage#onsuccess
+  const onSuccess = async (
+    data: SignMessageData,
+    variables: SignMessageVariables,
+    context: unknown
+  ) => {
+    await props.onSignSuccess({
+      recoveryType: RecoveryOptions.EOA,
+      signature: data,
+      identifier: variables.account as string,
+      challengeId: props.challengeData.id,
+    })
+  }
+
+  const onPress = async () => {
+    if (props.challengeData.challenge) {
+      signMessage(
+        {
+          message: props.challengeData.challenge,
```

Comment:
Let's clean up this challenge and convert to a 0x hex string. Also add a preamble to the challenge.
```suggestion
          message: byteaToHex(props.challengeData.challenge), 
```

Reply:
<img width="479" alt="Screenshot 2024-06-03 at 08 25 11" src="https://github.com/0xsend/sendapp/assets/95193764/f826d5a7-19c4-4164-9c82-cf69f46bd6fa">

Reply:
![image](https://github.com/0xsend/sendapp/assets/69170732/f261cd14-d58c-4884-8327-aa0832f15de7)